nixos/modules/services/logging/pushlog.nix

```
@@ -0,0 +1,171 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.pushlog;
+  unitType = types.submodule {
+    options = {
+      match = mkOption {
+        type = types.str;
+        default = ".*";
+      };
+      priorities = mkOption {
+        type = types.listOf types.int;
+        default = [
+          0
+          1
+          2
+          3
+          4
+          5
+          6
+        ];
+      };
+      include = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+      };
+      exclude = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+      };
+    };
+  };
+in
+{
+  options.services.pushlog = {
+    enable = mkEnableOption "Enable pushlog service to forward journald logs to Pushover";
+
+    package = mkPackageOption pkgs "pushlog" { };
+
+    environmentFile = mkOption {
+      type = with types; nullOr str;
+      description = lib.mdDoc ''
+        File containing the Pushover API credentials, in the
+        format of an EnvironmentFile as described by systemd.exec(5)
+
+        PUSHLOG_PUSHOVER_TOKEN, PUSHLOG_PUSHOVER_USER_KEY
+      '';
+      default = null;
+    };
+
+    settings = {
+      collect-timeout = mkOption {
+        type = types.int;
+        description = "Wait n seconds before sendings logs to bundle multiple messages";
+        default = 5;
+      };
+      deduplication-window = mkOption {
+        type = types.int;
+        description = "Remember messages for n minutes and avoid sending duplicates";
+        default = 30;
+      };
+      fuzzy-threshold = mkOption {
+        type = types.int;
+        description = "Use fuzzy matching with the given threshold (similarity in percent) to detect duplicates, set to 100 to disable";
+        default = 95;
+      };
+      title = mkOption {
+        type = with types; nullOr str;
+        description = "Optional title to use for all Pushover notifications";
+        default = null;
+      };
+      priority-map = mkOption {
+        type =
+          with types;
+          attrsOf (enum [
+            (-2)
+            (-1)
+            0
+            1
+            2
+          ]);
```

This is better expressed as:

```suggestion
          attrsOf (ints.between (-2) 2);
```